app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,36 @@
+class OrdersController < ApplicationController
+  before_action :authenticate_employee!
+
+  def index; end
+
+  def new
+    @reward = Reward.find(params[:reward])
+    if current_employee.earned_points < @reward.price
+      redirect_to rewards_path, notice: "This reward is too expensive for you.
+                                        You need #{@reward.price - current_employee.earned_points} points more."
+    else
+      render :new, locals: { order: Order.new }
+    end
+  end
+
+  def create
+    @reward = Reward.find(params[:order][:reward_id])
+    if current_employee.earned_points < @reward.price
+      redirect_to rewards_path, notice: "This reward is too expensive for you.
+                                          You need #{@reward.price - current_employee.earned_points} points more."
+    else
+      @order = Order.new(order_params)
+      if @order.save && current_employee.decrement(:earned_points, @reward.price).save
```

Comment:
This introduces a problem - what if `current_employee.decrement(:earned_points, @reward.price).save` fails? You end up with an already saved order. Think about using a transaction

Reply:
Okay, I get that. Thanks :) I have already implemented transaction into this method.